Source/Core/VideoCommon/Resources/TexturePool.h

```diff
@@ -0,0 +1,30 @@
+// Copyright 2025 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <unordered_map>
+
+#include "VideoCommon/TextureConfig.h"
+
+class AbstractTexture;
+
+namespace VideoCommon
+{
+class TexturePool
+{
+public:
+  TexturePool();
+  ~TexturePool();
+
+  void Reset();
+
+  std::optional<std::unique_ptr<AbstractTexture>> AllocateTexture(const TextureConfig& config);
```

Comment:
`std::optional<std::unique_ptr<AbstractTexture>>` is odd.
I think this can just be `std::unique_ptr<AbstractTexture>`.
At least one caller seems to not check that there's a value anyways.